File: solution files/C#/part-02/task-01-context-sensitivity/ContextExperiment.cs
using System.Collections.Generic;
using System.Linq;

public class ContextExperiment
{
    public List<string> ExtractNames(List<Dictionary<string, object>> data)
    {
        // Extract the 'name' field from each dictionary and return as a list of strings
        return data.Select(d => d["name"].ToString()).ToList();
    }
}

File: solution files/Javascript/part-02/task-01-context-sensitivity/contextExperiment.js
/**
 * @param {{ name: string }[]} data
 * @returns {string[]}
 */
function extractNames(data) {
    // Extract the 'name' field from each object and return as a list of strings
    return data.map((item) => item.name);
}

module.exports = { extractNames };
